pkg/egressgateway/fuzz_test.go

```diff
@@ -0,0 +1,39 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package egressgateway
+
+import (
+	"testing"
+
+	v2 "github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2"
+	v2alpha1 "github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2alpha1"
+	fuzz "github.com/AdaLogics/go-fuzz-headers"
+)
+
+func FuzzGetIfaceFirstIPv4Address(f *testing.F) {
+	f.Fuzz(func(t *testing.T, ifaceName string) {
+		_, _, _ = getIfaceFirstIPv4Address(ifaceName)
```

Comment:
This also seems to be a very thin layer over the underlying netlink library.

Reply:
I think it makes sense to have fuzzers for both the Cilium code and the 3rd-party library. I have modified this particular fuzzer to call earlier in the calltree. I have not rebased for ease of reviewing. LMKWYT

Reply:
A targeted fuzzer for the netlink calls have been added here: https://github.com/google/oss-fuzz/pull/9061/commits/61f755567253875b79608f87cec3d8dde053ca17

Reply:
What type of issue are we hoping to detect with these? The logic in the `Upsert()` and `ListRules()` functions is pretty trivial, mostly just making multiple calls into the underlying library. These are not user-facing APIs so the inputs are well controlled.

Reply:
@AdamKorcz and I discussed this out of band, and while we recognized that ideally this sort of testing would occur in the upstream library, there are no plans currently for adding fuzzers upstream. We can certainly pursue that, but in the mean time I am OK with merging these fuzzers as-is so that oss-fuzz can begin to run these fuzzers and gather information about the behaviour of these APIs. If upstream ends up integrating fuzzers on these APIs, then we can always retire these fuzzers at that point.